api/envoy/api/v2/core/config_source.proto

```diff
@@ -56,6 +60,17 @@ message ApiConfigSource {
 message AggregatedConfigSource {
 }
 
+// Rate Limit settings to be applied for discovery requests made by Envoy.
+message RateLimitSettings {
+  // Maximum number of tokens to be used for rate limting discovery request calls. If not set, a
+  // default value of 100 will be used.
+  google.protobuf.UInt32Value max_tokens = 1;
+
+  // Rate at which tokens will be filled per second. If not set, a default fill rate of 10 tokens
+  // per second will be used.
+  double fill_rate = 2;
```

Comment:
This needs to be a `google.protobuf.DoubleValue` to default to a non-zero value.